pylint/checkers/classes.py

```diff
@@ -913,29 +913,35 @@ def _check_unused_private_functions(self, node: nodes.ClassDef) -> None:
                     n.name for n in parent_scope.nodes_of_class(nodes.Name)
                 ):
                     continue
-            for attribute in node.nodes_of_class(nodes.Attribute):
-                if (
-                    attribute.attrname != function_def.name
-                    or attribute.scope() == function_def  # We ignore recursive calls
-                ):
-                    continue
-                if isinstance(attribute.expr, nodes.Name) and attribute.expr.name in (
-                    "self",
-                    "cls",
-                    node.name,
-                ):
-                    # self.__attrname
-                    # cls.__attrname
-                    # node_name.__attrname
+            for child in node.nodes_of_class((nodes.Name, nodes.Attribute)):
```

Comment:
```suggestion
            for child in node.nodes_of_class(nodes.Name):
                # Treat Name            
            for child in node.nodes_of_class(nodes.Attribute):
                # Treat Attribute
```
Would permit to remove some ``isinstance`` call but maybe I missed something.

Reply:
I thought about this but since `node.nodes_of_class()` iterates over all children I thought it might be better to keep this like this and only do an `isinstance`. I have not timed it myself, but iterating over all children twice versus some `isinstance`'s seemed worse.

Reply:
Hmm I would prefer not to do the performance heavier function twice...
@Pierre-Sassoulas opinion?

Reply:
Iterating on children is definitely worst than an isinstance